parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -128,8 +128,8 @@ import           Unison.Codebase.Causal         ( Causal
                                                 )
 import           Unison.Codebase.Path           ( Path(..) )
 import qualified Unison.Codebase.Path          as Path
-import           Unison.Codebase.NameSegment    ( NameSegment )
-import qualified Unison.Codebase.NameSegment   as NameSegment
+import           Unison.NameSegment             ( NameSegment )
+import qualified Unison.NameSegment            as NameSegment
```

Comment:
I moved `NameSegment` from `Unison.Codebase` to the `Unison` namespace and flipped the dependency between `Name` and `NameSegment` as well as `HashQualified'` and `NameSegment`.